compose: add docker-compose file for development

jsternberg opened the pull request:
This adds a simple docker compose file for development. The development
version of buildkit will build with the dockerfile, set debug to true,
use a tcp address to be compatible with the remote driver, and enable
the debug image through delve to allow for remote debugging.

It also configures additional services for help with development. At
the moment, this only includes jaeger for collecting traces but can also
include additional services in the future for things like metrics
collection.

tonistiigi commented on compose.yaml:
```diff
@@ -0,0 +1,33 @@
+
+configs:
+  buildkit_config:
+    file: ./dev/buildkitd.toml
```
where is this coming from?

tonistiigi reviewed (commented):
Because this is for dev can we move this under hack or contrib? Otherwise it might be confusing and some people could take it as way we recommend deploying buildkit.

jsternberg commented:
For the hack folder thing, is there a way to have `docker compose` work without using `-f` if the files are in hack?

jedevc reviewed (approved):
Would like some notes on how to use this, otherwise LGTM :tada:

Not quite sure where those notes belong, if they're long, could go into `docs/dev`, otherwise, could go directly into the comment for `hack/compose`.

Should we also add a disclaimer into `compose.yaml` that this isn't for general deployments? As mentioned in https://github.com/moby/buildkit/pull/4522#pullrequestreview-1804780452

jsternberg commented:
Added a comment to `hack/compose` with the usage and fixed the shell validation error. Also explained how to use the override file.

It might be useful to add some docs to it, but I think the comment is likely fine.

The usage is pretty simple. Just use `hack/compose` as an alternative to `docker compose`. All other arguments are forwarded except that `hack/compose` will pass the configuration file and the override if it's present.
